src/components/inputs/dropdown-input.tsx

```diff
@@ -0,0 +1,88 @@
+import { useTheme } from 'native-base';
+import React, { useState } from 'react';
+import { FlatList, Text, TouchableOpacity, View } from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import { DropdownInputProps, DropdownOptionProps } from '../../types';
+
+import { useDropdownInputStyles } from './input.styles';
+
+const DropdownInput: React.FC<DropdownInputProps> & {
+  Option: React.FC<DropdownOptionProps>;
+} = ({ children, disabled = false, label, onValueChange, selectedValue }) => {
+  const [isDropdownVisible, setDropdownVisible] = useState(false);
+  const dropdownStyles = useDropdownInputStyles();
+  const theme = useTheme();
+
+  const toggleDropdown = () => {
+    if (!disabled) {
+      setDropdownVisible(!isDropdownVisible);
+    }
+  };
+
+  const handleOptionSelect = (value: string) => {
+    onValueChange(value);
+    setDropdownVisible(false);
+  };
+
+  const inputTextColor = disabled ? theme.colors.secondary[500] : theme.colors.secondary[900];
+  const inputBgColor = disabled ? theme.colors.secondary[50] : theme.colors.white;
+
+  return (
+    <View style={dropdownStyles.wrapper}>
+      {label && <Text style={[dropdownStyles.label, { color: inputTextColor }]}>{label}</Text>}
+
+      <View style={dropdownStyles.container}>
+        <TouchableOpacity
+          activeOpacity={0.7}
```

Comment:
We want to control opacity via global theme so if an app decide to change it at one place, it gets change at all place. Inlining a 0.7 number is not ideal.